### ISSUE TITLE
CORS error

### ISSUE THREAD

**Issue by vaporpin** [kserve/kserve#1902]
/kind question

I am able to curl the inference service I set up using curl. However I'm not able to access it from my frontend development due to CORS error.

from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Usually the setting will be in Istios Virtual Service Setting using corsPolicy: allowOrigins. I tried to edit virtual service, sklearn-iris, however my changes will get overridden automatically (goes back to previous setting). In this case, how do I allow access from localhost or other host?

Thank you.

**Comment by ljtruong** [comment]
For anyone else having trouble with this, this was my workaround. 

1. Extend the predict handler like so.

```
import kserve.handlers as handlers
class CustomHandler(handlers.PredictHandler):
    def set_default_headers(self, *args, **kwargs):
        # modify as necessary
        self.set_header("Access-Control-Allow-Origin", "*") 
        self.set_header("Access-Control-Allow-Headers", "x-requested-with")
        self.set_header("Access-Control-Allow-Methods", "POST, GET, OPTIONS")
```

2. Update ModelServer class  `create_application` method. 

```
class CustomKServe(kserve.ModelServer):
    def create_application(self):
        return tornado.web.Application(
            [
                ...
                (r"/v1/models/([a-zA-Z0-9_-]+):predict", CustomHandler, dict(models=self.registered_models)),
                ...
            ],
            ....
        )
```

**Comment by WograTimKoehler** [comment]
Hello guys,
i just recently stumbled upon this issue again and got two possible fixes:  
1. Deploy ngnix as a reverse-proxy to handle CORS/Preflight requests.  
2. Or, and i think this would be preferable, add `CORSMiddleware` to the FastAPI app, created in `server.py`, like so:  
```py
class UvicornServer:
    def __init__(self, http_port: int, sockets: List[socket.socket],
                 data_plane: DataPlane, model_repository_extension, enable_docs_url,
                 log_config: Optional[Union[str, Dict]] = None,
                 access_log_format: Optional[str] = None):
        ...
        app = rest_server.create_application()
        ...
        app.add_middleware(
            CORSMiddleware,
            allow_origins=["*"],
            allow_credentials=True,
            allow_methods=["*"],
            allow_headers=["*"],
        )
```
(You can use it in your own `InferenceService` Image by building your own kserve wheel with `poetry build`)  

Is there a reason, why CORS Handling is not part of the source code, am i just blind and didn't see the documentation, or should i create a PR? I think using inference from a browser is pretty much a standard usecase.  

In any case, thank you very much for your work and i hope this answer helps anyone with the same issue.  

Edit:
Just found this PR. I guess this would pretty much solve the issue https://github.com/kserve/kserve/pull/2954

**Comment by elukey** [comment]
Found this issue recently, I think that the best solution would be #721 (if people use istio as proxy/sidecar of course).
